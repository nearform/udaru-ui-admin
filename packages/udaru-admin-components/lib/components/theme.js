// Adapted from default theme on https://github.com/jxnblk/styled-system

// breakpoint values
// any array length works with styled-system
const breakpoints = ['500px', '800px', '1200px']

const colors = {
  primary: '#337ab7',
  primaryActive: '#02A171',
  danger: '#d9534f',
  lightGrey: '#D8D8D8',
  lightGrey2: '#888',
  lightGrey3: '#eee',
  darkGrey: '#555',
  black: '#333',
  green: '#5cb85c',
  lightBlue: '#5bc0de',
  orange: '#f0ad4e'
}

export { colors }

export const radius = '4px'

export const maxContainerWidth = '1280px'

// boxShadows
export const boxShadows = [
  `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
]

export const speeds = {
  slow: '1s',
  medium: '.4s',
  fast: '.2s'
}

export const easings = {
  easeOut: 'cubic-bezier(0,.6,.3,1)',
  expoEaseOut: 'cubic-bezier(0,1,.3,1)',
  bounce: 'cubic-bezier(0,1.2,.5,1.4)'
}

// space is used for margin and padding scales
// it's recommended to use powers of two to ensure alignment
// when used in nested elements
// numbers are converted to px
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512]

// typographic scale
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 96, 128]

// for any scale, either array or objects will work
const lineHeights = [1, 1.125, 1.25, 1.5]

const fontWeights = {
  normal: 500,
  bold: 700
}

const letterSpacings = {
  normal: 'normal',
  caps: '0.25em'
}

// border-radius
const radii = [0, 2, 4, 8]

const borders = [0, '1px solid', '2px solid']

const shadows = [`0 1px 2px 0 ${colors.text}`, `0 1px 4px 0 ${colors.text}`]

const theme = {
  breakpoints,
  colors,
  space,
  fontSizes,
  lineHeights,
  fontWeights,
  letterSpacings,
  radii,
  radius,
  borders,
  shadows,
  boxShadows,
  speeds,
  easings
}

export default theme
