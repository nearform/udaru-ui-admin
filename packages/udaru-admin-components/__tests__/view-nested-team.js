import React from 'react'
import renderer from 'react-test-renderer'
import ViewNestedTeam from 'view-nested-team'
import { createSerializer } from 'jest-emotion'
import * as emotion from 'emotion'
import theme from '../lib/components/theme'

expect.addSnapshotSerializer(createSerializer(emotion))

jest.mock('react-bootstrap-table', () => {
  return {
    BootstrapTable: props => {
      return (
        <table>
          {' // mocked table output'}
          {props.children}
        </table>
      )
    },
    TableHeaderColumn: props => {
      return (
        <tr>
          <th>{props.dataField + ' // mocked table header column'}</th>
        </tr>
      )
    }
  }
})

it('should render with default props', async () => {
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: true,
          json: function() {
            return {
              data: [],
              total: 0
            }
          }
        })
      })
  )
  const component = renderer.create(<ViewNestedTeam theme={theme} />)
  const instance = component.root.instance
  await instance.componentDidMount()

  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()

  global.fetch.mockRestore()
})

it('should load data successfully', async () => {
  const data = [{ id: 1 }]
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: true,
          json: function() {
            return {
              data,
              total: 1
            }
          }
        })
      })
  )
  const component = renderer.create(
    <ViewNestedTeam
      theme={theme}
      udaruUrl="my-udaru-url"
      authorization="my-authorization"
      org="my-org"
      parentTeamId="1"
      currentPage={1}
      limit="5"
    />
  )
  const instance = component.root.instance

  expect(instance.state.loading).toBeTruthy()
  await instance.componentDidMount()

  expect(global.fetch.mock.calls[0][0]).toEqual(
    'my-udaru-url/authorization/teams/1/nested?page=1&limit=5'
  )
  expect(global.fetch.mock.calls[0][1]).toEqual({
    headers: {
      authorization: 'my-authorization',
      org: 'my-org'
    }
  })
  expect(instance.state.loading).toBeFalsy()
  expect(instance.state.data).toBe(data)
  expect(instance.state.total).toBe(1)
  expect(instance.state.error).toBeNull()

  global.fetch.mockRestore()
})

it('should handle error when loading data', async () => {
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: false,
          json: function() {
            return {}
          }
        })
      })
  )
  const component = renderer.create(
    <ViewNestedTeam
      theme={theme}
      udaruUrl="my-udaru-url"
      authorization="my-authorization"
      org="my-org"
      parentTeamId="1"
      currentPage={1}
      limit="5"
    />
  )
  const instance = component.root.instance

  expect(instance.state.loading).toBeTruthy()
  await instance.componentDidMount()

  expect(instance.state.loading).toBeFalsy()
  expect(instance.state.data).toEqual([])
  expect(instance.state.total).toBe(0)
  expect(instance.state.error).toBeDefined()
  expect(instance.state.error.message).toBe('there was an error loading team.')

  global.fetch.mockRestore()
})

it('should cancel all promises running when component is unmounted', async () => {
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: false,
          json: function() {
            return {}
          }
        })
      })
  )
  const component = renderer.create(
    <ViewNestedTeam
      theme={theme}
      udaruUrl="my-udaru-url"
      authorization="my-authorization"
      org="my-org"
      parentTeamId="1"
      currentPage={1}
      limit="5"
    />
  )
  const instance = component.root.instance

  await component.unmount()

  const { _runningPromises } = instance

  expect(_runningPromises.every(p => p.hasCanceled())).toBeTruthy()

  global.fetch.mockRestore()
})
