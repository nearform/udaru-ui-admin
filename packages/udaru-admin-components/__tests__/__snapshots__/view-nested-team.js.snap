// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with default props 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "padding": "60px 20px",
    }
  }
>
  <div>
    <h2
      style={
        Object {
          "margin": 0,
        }
      }
    >
      Nested Teams for 
    </h2>
  </div>
  <table>
     // mocked table output
    <tr>
      <th>
        view // mocked table header column
      </th>
    </tr>
    <tr>
      <th>
        id // mocked table header column
      </th>
    </tr>
    <tr>
      <th>
        name // mocked table header column
      </th>
    </tr>
    <tr>
      <th>
        description // mocked table header column
      </th>
    </tr>
    <tr>
      <th>
        usersCount // mocked table header column
      </th>
    </tr>
  </table>
</div>
`;
